lib/shoulda/matchers/active_record/have_db_index_matcher.rb

```diff
@@ -197,16 +197,28 @@ def correct_unique?
         def matched_index
           @_matched_index ||=
             if expected_columns.one?
-              actual_indexes.detect do |index|
+              sorted_indexes.detect do |index|
                 Array.wrap(index.columns) == expected_columns
               end
             else
-              actual_indexes.detect do |index|
+              sorted_indexes.detect do |index|
                 index.columns == expected_columns
               end
             end
         end
 
+        def sorted_indexes
+          indexes = actual_indexes
+
+          if qualifiers.include?(:unique)
+            indexes.sort_by! do |index|
+              index.unique == qualifiers[:unique] ? 0 : 1
+            end
+          end
+
+          indexes
```

Comment:
The bang at the end of `sort_by` modifies the array in place. In-place mutation tends to be useful for performant or complex code, however, given how simple this method is, I wonder if we say things a bit more straightforwardly. What do you think about this?

``` suggestion
          if qualifiers.include?(:unique)
            actual_indexes.sort_by do |index|
              index.unique == qualifiers[:unique] ? 0 : 1
            end
          else
            actual_indexes
          end
```

Reply:
Sure. I'd done it using the `sort_by!` in case there were any new qualifiers added to support applying other sorting to the list. But it can of course be refactored if/when that's the case! Will update.